### PR TITLE
Cycling through match errors

### PR DESCRIPTION
**Pull request by 5n8ke**
Instead of only allowing to jump to the first location found by the given errorMatch regular expression, you can cycle through every error with build:error-match. At the end it loops around to the beginning.
build:error-match-first (cmd-alt-h) jumps directly back to the beginning.

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -135,27 +137,63 @@ module.exports = {
     return value;
   },
 
+  errorParse: function() {
+    this.match = [];
+    var regex = XRegExp(this.cmd.errorMatch);
```
Comment (5n8ke) [inline lib/build.js]: Depending on the regex used the multiline flag possibly needs to be added. But with quick and dirty regex for g++ it is not needed.

**Comment by noseglid** [comment]
This looks great. Would you mind adding some tests to this?

**Comment by 5n8ke** [comment]
I hope these tests are fine, it's the first time creating some for me. It tests cycling and wrap around and jumping to the first error match.

I don't know why travis failed. In my own repo the build and test are without error.

**Comment by noseglid** [comment]
I just re-started the build. Seems as there was an outage in the travis service.

**Comment by noseglid** [comment]
Thanks you very much for this contribution. It will make an excellent addition to atom-build!
It will be included in the next version (`v0.29.0`).

**Pull request merged**